pkgpanda/build/__init__.py

```diff
@@ -449,13 +450,52 @@ def hash_checkout(item):
         raise NotImplementedError("{} of type {}".format(item, type(item)))
 
 
-def hash_folder(directory):
+def hash_files_in_folder(directory):
+    assert not directory.startswith('/'), \
+        "For the hash to be reproducible on other machines relative paths must always be used. " \
+        "Got path: {}".format(directory)
+    directory = directory.rstrip('/')
     file_hash_dict = {}
+    # TODO(cmaloney): Disallow symlinks as they're hard to hash, people can symlink / copy in their
+    # build steps if needed.
     for root, dirs, filenames in os.walk(directory):
+        assert not root.startswith('/')
```

Comment:
Just a nit: I am wondering if there are os.walk implementations that will give you absolute paths for non-absolute argument.